build-support/scripts/envoy-library-references.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```

Comment:
Some explanation of why this is necessary, and some breadcrumbs back to the original issue would be great. I imagine we will forget why this is necessary and explaining here in the script that generates it would probably be a good place.

Some mention of the call to `proto.RegisterFile` in package level `init()` functions would be a good clue.